### ISSUE TITLE
Deployment is not started automatically in 1.5.1

### ISSUE THREAD

**Issue by kargakis** [openshift/origin#15136]
This is not a problem in v1.6 (will need to be cherry-picked though), but in 1.5.1 it seems that an imported image never triggers a deployment if the resulting pullspec in the container is already set before creation but is missing from the lastTriggeredImage field in the triggers (the field is zero on creation). This is problematic because the trigger is never resolved, and deployment is blocked.

@mfojtik @tnozicka 

<details>

```
[mkargaki@dhcp129-211 origin]$ oc cluster up --version=v1.5.1
Starting OpenShift using openshift/origin:v1.5.1 ...
OpenShift server started.

The server is accessible via web console at:
    https://127.0.0.1:8443

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin

[mkargaki@dhcp129-211 origin]$ oc process -f https://raw.githubusercontent.com/nhr/znc-cluster-app/master/znc-template-basic.yaml | oc create -f -
imagestream "znc-cluster-app" created
deploymentconfig "znc-cluster-app" created
service "znc-cluster-app" created
route "znc-cluster-app" created
[mkargaki@dhcp129-211 origin]$ oc get dc
NAME              REVISION   DESIRED   CURRENT   TRIGGERED BY
znc-cluster-app   0          1         0         config,image(znc-cluster-app:latest)
[mkargaki@dhcp129-211 origin]$ oc get po
No resources found.
[mkargaki@dhcp129-211 origin]$ oc get dc -o yaml
apiVersion: v1
items:
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    annotations:
      openshift.io/generated-by: OpenShiftNewApp
    creationTimestamp: 2017-07-11T13:24:48Z
    generation: 1
    labels:
      app: znc-cluster-app
    name: znc-cluster-app
    namespace: myproject
    resourceVersion: "1000"
    selfLink: /oapi/v1/namespaces/myproject/deploymentconfigs/znc-cluster-app
    uid: 5039b6c6-663c-11e7-ae2f-68f72840e7df
  spec:
    replicas: 1
    selector:
      app: znc-cluster-app
      deploymentconfig: znc-cluster-app
    strategy:
      activeDeadlineSeconds: 21600
      resources: {}
      rollingParams:
        intervalSeconds: 1
        maxSurge: 25%
        maxUnavailable: 25%
        timeoutSeconds: 600
        updatePeriodSeconds: 1
      type: Rolling
    template:
      metadata:
        annotations:
          openshift.io/container.znc-cluster-app.image.entrypoint: '["/opt/znc-run/znc_runner.sh"]'
          openshift.io/generated-by: OpenShiftNewApp
        creationTimestamp: null
        labels:
          app: znc-cluster-app
          deploymentconfig: znc-cluster-app
      spec:
        containers:
        - image: nhripps/znc-cluster-app@sha256:cf496fa3969f044a3cbc1d43042adacb4f6653500922f65b8dd06178a9523acd
          imagePullPolicy: Always
          name: znc-cluster-app
          ports:
          - containerPort: 6697
            protocol: TCP
          resources: {}
          terminationMessagePath: /dev/termination-log
        dnsPolicy: ClusterFirst
        restartPolicy: Always
        securityContext: {}
        terminationGracePeriodSeconds: 30
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - znc-cluster-app
        from:
          kind: ImageStreamTag
          name: znc-cluster-app:latest
          namespace: myproject
      type: ImageChange
  status:
    availableReplicas: 0
    conditions:
    - lastTransitionTime: 2017-07-11T13:24:48Z
      lastUpdateTime: 2017-07-11T13:24:48Z
      message: Deployment config does not have minimum availability.
      status: "False"
      type: Available
    latestVersion: 0
    observedGeneration: 1
    replicas: 0
    unavailableReplicas: 0
    updatedReplicas: 0
kind: List
metadata: {}
resourceVersion: ""
selfLink: ""
[mkargaki@dhcp129-211 origin]$ oc get is
NAME              DOCKER REPO                                 TAGS      UPDATED
znc-cluster-app   172.30.1.1:5000/myproject/znc-cluster-app   latest    53 seconds ago
[mkargaki@dhcp129-211 origin]$ oc rollout latest dc/znc-cluster-app
Error from server (BadRequest): cannot trigger a deployment for "znc-cluster-app" because it contains unresolved images
```
</details>

**Comment by kargakis** [comment]
@nhr I think you are the lucky winner of this bug, this seems like a rare case where the image that is downloaded is already set in the template, [so we ignore doing the update in that case](https://github.com/openshift/origin/blob/6c5dafb9789eb49a4422a6ca6b4112d43288e4a6/pkg/deploy/registry/instantiate/rest.go#L179) (obviously a bug).

**Comment by nhr** [comment]
@kargakis I love winning things! :-) Is there a change that I can make to my [template](https://raw.githubusercontent.com/nhr/znc-cluster-app/master/znc-template-basic.yaml) in order to work around this bug?

**Comment by kargakis** [comment]
Yes, you can set the image inside your container to something other than the actual hash of the image, for example:
<details>

```
apiVersion: v1
kind: Template
metadata:
  creationTimestamp: null
  name: znc-cluster-app
objects:
- apiVersion: v1
  kind: ImageStream
  metadata:
    annotations:
      openshift.io/generated-by: OpenShiftNewApp
      openshift.io/image.dockerRepositoryCheck: 2016-08-18T18:49:29Z
    creationTimestamp: null
    generation: 2
    labels:
      app: znc-cluster-app
    name: znc-cluster-app
  spec:
    tags:
    - from:
        kind: DockerImage
        name: nhripps/znc-cluster-app
      name: latest
- apiVersion: v1
  kind: DeploymentConfig
  metadata:
    annotations:
      openshift.io/generated-by: OpenShiftNewApp
    creationTimestamp: null
    generation: 2
    labels:
      app: znc-cluster-app
    name: znc-cluster-app
  spec:
    replicas: 1
    selector:
      app: znc-cluster-app
      deploymentconfig: znc-cluster-app
    strategy:
      resources: {}
      rollingParams:
        intervalSeconds: 1
        maxSurge: 25%
        maxUnavailable: 25%
        timeoutSeconds: 600
        updatePeriodSeconds: 1
      type: Rolling
    template:
      metadata:
        annotations:
          openshift.io/container.znc-cluster-app.image.entrypoint: '["/opt/znc-run/znc_runner.sh"]'
          openshift.io/generated-by: OpenShiftNewApp
        creationTimestamp: null
        labels:
          app: znc-cluster-app
          deploymentconfig: znc-cluster-app
      spec:
        containers:
        - image: nhripps/znc-cluster-app
          imagePullPolicy: Always
          name: znc-cluster-app
          ports:
          - containerPort: 6697
            protocol: TCP
          resources: {}
          terminationMessagePath: /dev/termination-log
        dnsPolicy: ClusterFirst
        restartPolicy: Always
        securityContext: {}
        terminationGracePeriodSeconds: 30
    test: false
    triggers:
    - type: ConfigChange
    - imageChangeParams:
        automatic: true
        containerNames:
        - znc-cluster-app
        from:
          kind: ImageStreamTag
          name: znc-cluster-app:latest
          namespace: myproject
      type: ImageChange
- apiVersion: v1
  kind: Service
  metadata:
    annotations:
      openshift.io/generated-by: OpenShiftNewApp
    creationTimestamp: null
    labels:
      app: znc-cluster-app
    name: znc-cluster-app
  spec:
    ports:
    - name: 6697-tcp
      port: 6697
      protocol: TCP
      targetPort: 6697
    selector:
      app: znc-cluster-app
      deploymentconfig: znc-cluster-app
    sessionAffinity: None
    type: ClusterIP
- apiVersion: v1
  kind: Route
  metadata:
    annotations:
      openshift.io/host.generated: "true"
    creationTimestamp: null
    labels:
      app: znc-cluster-app
    name: znc-cluster-app
  spec:
    port:
      targetPort: 6697-tcp
    tls:
      termination: passthrough
    to:
      kind: Service
      name: znc-cluster-app
      weight: 100
```
</details>

**Comment by nhr** [comment]
Awesome, thanks @kargakis !

**Comment by kargakis** [comment]
Fixed in Origin 1.6 and OSE 3.5. Merges in the release-1.5 branch are badly broken and it's unlike that the fix for Origin 1.5 (https://github.com/openshift/origin/pull/15139) will land anytime soon.
